prov/efa/configure.m4

```diff
@@ -113,6 +114,46 @@ AC_DEFUN([FI_EFA_CONFIGURE],[
 	AC_SUBST(efa_CPPFLAGS)
 	AC_SUBST(efa_LDFLAGS)
 	AC_SUBST(efa_LIBS)
+
+	AC_ARG_ENABLE([efa-unit-test],
+		[AS_HELP_STRING([--enable-efa-unit-test=CMOCKA_INSTALL_DIR],
+				[Provide a path to the CMocka installation directory
+				 in order to enable EFA Unit Tests.])])
+
+	cmocka_dir=""
+	if [test x"$enable_efa_unit_test" = x"yes"]; then
+    	cmocka_dir=""
+	else
+   		cmocka_dir="$enable_efa_unit_test"
+	fi
+
+	if [ test x"$enable_efa_unit_test" != x"" ]; then
+		FI_CHECK_PACKAGE(cmocka,
+						 [cmocka.h],
+						 [cmocka],
+						 [_expect_any],
+						 [],
+						 [$cmocka_dir],
+						 [$cmocka_dir/lib64],
+						 [
+							AC_DEFINE([EFA_UNIT_TEST], [1], [EFA unit testing])
+							efatest_LIBS=" -lcmocka -libverbs -static $efatest_LIBS"
+							efa_LIBS+=" $cmocka_LDFLAGS -lcmocka -libverbs -static $efatest_LIBS"
+						 ],
+					 	 [
+							AC_DEFINE([EFA_UNIT_TEST], [0], [EFA unit testing])
+							AC_MSG_ERROR([Cannot compile EFA unit tests without a valid Cmocka installation directory.])
+							efa_unit_test=no
+					 	 ],
+						 [
+							#include <stdarg.h>
+							#include <stddef.h>
+							#include <setjmp.h>
+					 	 ])
+	fi
```

Comment:
Do not you need a `else` branch and to define `EFA_UNIT_TEST` to 0?

Reply:
It compiles without it, but I added it in for explicitness